modules/src/applications/transfer/acknowledgement.rs

```diff
@@ -1,26 +1,39 @@
+use core::fmt::{Display, Formatter};
+
+use serde::{Deserialize, Serialize};
+
 use super::error::Error;
 use crate::core::ics26_routing::context::Acknowledgement as AckTrait;
 use crate::prelude::*;
-use core::fmt::{Display, Formatter};
-
-use serde::{Deserialize, Deserializer};
 
 /// A string constant included in error acknowledgements.
 /// NOTE: Changing this const is state machine breaking as acknowledgements are written into state
 pub const ACK_ERR_STR: &str = "error handling packet on destination chain: see events for details";
-pub const ACK_SUCCESS_B64: &[u8] = b"AQ==";
 
-#[derive(Clone, Debug)]
+/// A successful acknowledgement, equivalent to `base64::encode(0x01)`.
+pub const ACK_SUCCESS_B64: &str = "AQ==";
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub enum ConstAckSuccess {
+    #[serde(rename = "AQ==")]
+    Success,
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "lowercase")]
 pub enum Acknowledgement {
-    /// Equivalent to b"AQ==" (i.e. `base64::encode(0x01)`)
-    Success(Vec<u8>),
+    /// Successful Acknowledgement
+    /// e.g. `{"result":"AQ=="}`
+    #[serde(rename = "result")]
+    Success(ConstAckSuccess),
     /// Error Acknowledgement
+    /// e.g. `{"error":"cannot unmarshal ICS-20 transfer packet data"}`
```

Comment:
Can we use `#[serde(rename = "error")]` and remove `#[serde(rename_all = "lowercase")]` on the enum to make things a bit more explicit?

Reply:
Done! 85f361000aa48847e7a022336f315c012d24ba88